### DISCUSSION TITLE
ESP32 image built twice?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#982] [General]
Caution: I'm still new here... 🤪
I installed the moddable SDK
I installed ESP-IDF
I ran the Hello World example on a NodeMCU: Yay! 🚀
I looked at the build log and ... it seems to me like the flash image is built twice? (As if it didn't already take long enough.) I see two times:
```
[100%] Built target gen_project_binary                                            
xs_esp32.bin binary size 0x5ba10 bytes. Smallest app partition is 0x390000 bytes. 0x3345f0 bytes (90%) free.                                                        
[100%] Built target app_check_size                                                
[100%] Built target app                                                           
```
I haven't done a raw esp-idf build in a while, so maybe there's something obvious I'm missing...
The full build output (long) is attached (I ran `mcconfig -d -m -p esp32/nodemcu`).
[log.txt](https://github.com/Moddable-OpenSource/moddable/files/10221264/log.txt)

**Comment by andycarle** [comment]
Agreed that something looks off there. It almost looks like you have two builds happening at the same time, interleaved in your log output...

This bit is particularly suspicious:

```console
Project build complete. To flash, run this command:
/home/tve/.espressif/python_env/idf4.4_py3.10_env/bin/python ../../../../../../../../esp-idf/components/esptool_py/esptool/esptool.py -p (PORT) -b 460800 --before default_reset --after hard_reset --chip esp32  write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x1000 build/bootloader/bootloader.bin 0x8000 build/partition_table/partition-table.bin 0x10000 build/xs_esp32.bin
or run 'idf.py -p (PORT) flash'
Executing action: flash
Serial port /dev/ttyUSB0
Connecting....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting.......
Detecting chip type...[  0%] Built target _project_elf_src
[  0%] Built target memory_ld
```

That's showing that the build has completed and it wants to flash, but then it just keeps on building. Odd. I'll take a look

**Reply by tve** [reply]
I built a docker image based on debian bullseye and I see the same double-build.
